pkgs/checks/lib/src/checks.dart

```diff
@@ -207,6 +207,18 @@ abstract class Context<T> {
   /// this method will throw.
   Future<Check<R>> nestAsync<R>(
       String label, FutureOr<Extracted<R>> Function(T) extract);
+
+  /// Report that a check failed after the condition has already succeeded.
+  ///
+  /// A condition that some event _never_ happens will not have any point at
+  /// which it can be considered "complete", so a rejection may occur at any
+  /// time.
+  ///
+  /// Late rejections are ignored in [softCheck] and [softCheckAsync], so a
```

Comment:
I'm not sure if this is worth trying to eagerly block. We currently throw if you try to use an async expectation in `softCheck` instead of `softCheckAsync`, so we could potentially have another type of interaction which isn't allowed from either of those, but is allowed from `checkThat`.

Reply:
I found a pattern that I don't think it too ugly. I'll run with that since I think it will be less confusing to have something which loudly fails rather than silently does nothing.